src/sentry/eventstore/models.py

```diff
@@ -372,6 +372,16 @@ def get_hashes(self, force_config=None) -> CalculatedHashes:
             hierarchical_variants
         )
 
+        if flat_hashes:
+            sentry_sdk.set_tag("event.get_hashes.first_flat_variant_name", flat_hashes[0][0])
+        if hierarchical_hashes:
+            sentry_sdk.set_tag(
+                "event.get_hashes.first_hierarchical_variant_name", hierarchical_hashes[0][0]
+            )
+
+        flat_hashes = [hash_ for _, hash_ in flat_hashes]
+        hierarchical_hashes = [hash_ for _, hash_ in hierarchical_hashes]
```

Comment:
Why not expose the names in the interface of `get_hashes`, and then call `set_tag` in the event_manager? Just so this tag is not set accidentally when `get_hashes()` is called from other places.

Reply:
I think we can live with this tag being accidentally set. Since the majority of get_hashes callers are untyped I didn't dare refactor this.